tilings/algorithms/gridded_perm_reduction.py

```diff
@@ -124,6 +125,28 @@ def minimal_reqs(
                 return (tuple(),)
         return tuple(minimized_requirements)
 
+    def remove_redundant(self, requirements: List[Requirement]) -> List[Requirement]:
+        relevant_reqs: List[Requirement] = []
+        for idx, requirement in enumerate(requirements):
```

Comment:
"requirement" here is a requirement list?

Reply:
yes, its type is precisely is `Tuple[GriddedPerm, ...]`